src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -295,12 +321,19 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
         [SupportedOSPlatform("windows")]
         public static IntPtr /* IUnknown* */ GetIUnknownForObject(object o)
         {
-            if (o is null)
+            if (Marshal.IsComSupported)
```

Comment:
This API is supported with ComWrappers. It should be disabled for just the built-in COM case in the VM.

Reply:
You may want to check that the tests under `SRC\tests\Interop\COM\ComWrappers` work even when COM is disabled.